proposals/0008-non-member-operator-overloading.md

```diff
@@ -44,6 +44,53 @@ under HLSL 2021 as an extension. Using the feature in HLSL 2021 mode as an
 extension will produce warnings so that users can be aware of portability issues
 that may arise between compiler versions.
 
+## Detailed Design
+
+### **Overloaded Operators** Spec Language
+
+```latex
+\begin{grammar}
+  \define{operator-function-id}\br
+    \terminal{operator} operator\br
+
+  \define{operator} one of\br
+    \terminal{+} \terminal{-} \terminal{*} \terminal{/} \terminal{\%}
+    \terminal{\^{}} \terminal{\&} \terminal{|} \terminal{\~{}} \terminal{!}
+    \terminal{=} \terminal{<} \terminal{>}\br
+    \terminal{+=} \terminal{-=} \terminal{*=} \terminal{/=} \terminal{\%=}
+    \terminal{\^{}=} \terminal{\&=} \terminal{|=}\br
+    \terminal{<<} \terminal{>>} \terminal{>>=} \terminal{<<=}
+    \terminal{==} \terminal{!=} \terminal{<=} \terminal{>=} \terminal{\&\&}
+    \terminal{||}\br
+    \terminal{++} \terminal{--} \terminal{,} \terminal{()}
+    \terminal{\[\]}\br
+\end{grammar}
+```
+![Latex Rendering](0008-assets/OperatorGrammar.png)
+
+An _operator function_ is a function declared with an _operator-function-id_ as
+its name. An _operator function template_ is a function template
+operator-function-id as its name.
+
+Both the binary and unary forms of `+`, `-`, `*` and `&` can be overloaded.
```

Comment:
It might be worth spending a sentence on how `operator-function-id`s are determined to be binary or unary. (I assume its implicit in the number of parameters?)